### PR TITLE
PSQLODBC Datatype Test for sys.bit

### PR DESCRIPTION
**Pull request by ColinKYuen**
### Description

Adds ODBC Datatype Test for `sys.bit`
- [x] Create Table
- [x] Insert valid values
- [x] Insert invalid values
- [x] Update table with valid values
- [x] Update table with invalid values
- [x] Bitwise operations
- [x] Table View
- [x] Unique Constraint
- [x] Composite Key

### Problems
- Bitwise operators do not seem to be working on both ODBC tests nor connecting via DBeaver
- Actual datatype name and length do not match expected (Can be seen in `TableCreation` test)

### Issues Resolved

[FISH55](https://bitquill.atlassian.net/browse/FISH-55)

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/odbc/test_pg_datatype_bit.cpp [hunk 1]
```diff
@@ -0,0 +1,581 @@
+#include <gtest/gtest.h>
```
Comment (ghost) [inline test/odbc/test_pg_datatype_bit.cpp]: Looks good. There are a few more test cases that were called out by Amazon as good to have:

> Determining if primary key or unique constraint exists on columns that support usage a the constraint type. Also test multi-column PK with mixed types like: int/smallint, int/varchar, int/datetime

Can these also be added, if possible?

File: test/odbc/test_pg_datatype_bit.cpp [hunk 2]
```diff
@@ -0,0 +1,581 @@
...
+  OdbcHandler odbcHandler;
+
+  vector <string> valid_inserted_values = {
+    "0",
+    "1",
+    "-0",
+    "-1",
+    "2",
+    "NULL"
+  };
```
Comment (ghost) [inline test/odbc/test_pg_datatype_bit.cpp]: Maybe also have a bigger number, like `-24` or `123`?

File: test/odbc/test_pg_datatype_bit.cpp (hunk 1 above)
Comment (ColinKYuen) [reply test/odbc/test_pg_datatype_bit.cpp]: Is there a document I can reference for any additional test cases? 

For now I'll look into the one listed here after I finish up some other things.

File: test/odbc/test_pg_datatype_bit.cpp [hunk 3]
```diff
@@ -0,0 +1,764 @@
...
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("VIEW", VIEW_NAME));
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_Bit, Table_Unique_Constraints) {
```
Comment (ghost) [inline test/odbc/test_pg_datatype_bit.cpp]: Sorry, one more thing about these UNIQUE & PRIMARY KEY column tests. Is it possible to check if these columns are still UNIQUE & PRIMARY KEY after table creation? 

I think using the ODBC functions [SQLSpecialColumns](https://www.ibm.com/docs/en/db2-for-zos/11?topic=functions-sqlspecialcolumns-get-special-row-identifier-columns) & [SQLPrimaryKeys](https://www.ibm.com/docs/en/db2-for-zos/11?topic=functions-sqlprimarykeys-get-primary-key-columns-table) will be useful in this case.

There are examples of their use in the `odbc/test_metadata.cpp` file.

**Comment by ColinKYuen** [comment]
PR made to upstream.

**Pull request closed without merging**